react-basic-practice/src/components/CartBar/index.jsx

```diff
@@ -2,57 +2,83 @@ import React from 'react';
 import Button from '../common/Button';
 import CartItem from '../CartItem';
 import closeButton from '@/assets/images/iconButton/btn-close.png';
+import { useCart } from '@/contexts/CartProvider';
+import { useToast } from '@/contexts/ToastProvider';
 
 import './index.css';
+import Toast from '../Toast';
 
-const data = [
-  {
-    idProduct: 2,
-    quantity: 3,
-    name: 'hal earrings',
-    price: 25,
-  },
-  {
-    idProduct: 1,
-    quantity: 4,
-    name: 'lira earrings',
-    price: 20,
-  },
-];
-
-const CartBar = () => {
-  // const {
-  //   onClose, data,
-  // } = props;
-  const onDeleteCartItem = () => {};
-  const onHandleCloseNav = () => {
-    // onClose(false);
+const CartBar = (props) => {
+  const { handleCloseCart } = props;
+  const { listItem, setListItem, updateItemCart } = useCart();
+  const { toast, setToast } = useToast();
+
+  const onDeleteCartItem = (e) => {
+    try {
+      const idSelected = e.target.closest('.cart-item').id;
```

Comment:
Please use this: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes